cli/src/complete.rs

```diff
@@ -271,6 +273,67 @@ pub fn operations() -> Vec<CompletionCandidate> {
     })
 }
 
+fn config_keys_rec(
+    prefix: ConfigNamePathBuf,
+    properties: &serde_json::Map<String, serde_json::Value>,
+    acc: &mut Vec<CompletionCandidate>,
+    only_leaves: bool,
+) {
+    for (key, value) in properties {
+        if key == "type" {
+            // used by "additionalProperties", ignore
+            continue;
+        }
+        let mut prefix = prefix.clone();
+        prefix.push(key);
+
+        let value = value.as_object().unwrap();
+        match value.get("type").and_then(|v| v.as_str()) {
+            Some("object") => {
+                if !only_leaves {
+                    let escaped_key = prefix.to_string();
+                    acc.push(CompletionCandidate::new(escaped_key));
+                }
+                let Some(properties) = value.get("properties") else {
+                    continue;
```

Comment:
nit: might be better to suggest `<prefix>." if no child properties are defined. (can be added later)

Reply:
You mean with a trailing dot? These non-leaf keys are currently only suggested for `config list`, where omitting the dot makes sense.

It would make sense to suggest the trailing dot for the other commands, though. I agree that this can be added later.

Reply:
> You mean with a trailing dot?

Yes. I think it's nice if `aliases.` (without trailing space) is suggested for `jj config set a<TAB>`, but I'm not sure if that's doable.